pyccel/parser/semantic.py

```diff
@@ -503,6 +503,17 @@ def _infere_type(self, expr, **settings):
             d_var['precision'     ] = var.precision
             return d_var
 
+        elif isinstance(expr, IndexedElement):
+            d_var['datatype'   ] = expr.dtype
+            d_var['memory_handling'] = 'heap' if expr.rank > 0 else 'stack'
+            d_var['shape'      ] = expr.shape
+            d_var['rank'       ] = expr.rank
+            d_var['order'      ] = expr.order
+            d_var['precision'  ] = expr.precision
+            d_var['cls_base'   ] = expr.base.cls_base
+            d_var['is_tuple'   ] = expr.base.is_tuple
```

Comment:
Couldn't this simply be ` d_var['is_tuple'   ] = isinstance(expr.base, TupleVariable)`?